pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,178 @@
+
+	cmd.SetDescription("apply", "Setting up GitOps and apply a Quick Start profile", "")
+
+	var opts options
+
+	cmd.SetRunFuncWithNameArg(func() error {
+		return doApplyGitops(cmd, opts)
+	})
+
+	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		fs.StringVarP(&opts.quickstartNameArg, "quickstart-profile", "", "", "name or URL of the Quick Start profile. For example, app-dev.")
+		fs.StringVarP(&opts.git.URL, "git-url", "", "", "URL for the git repository that will contain the cluster components")
+		fs.StringVarP(&opts.git.Branch, "git-branch", "", "master", "Git branch")
+		fs.StringVarP(&opts.outputPath, "output-path", "", "./", "Path to directory where the GitOps repo will be cloned")
+		fs.StringVar(&opts.git.User, "git-user", "Flux", "Username to use as Git committer")
+		fs.StringVar(&opts.git.Email, "git-email", "", "Email to use as Git committer")
+		fs.StringVar(&cfg.Metadata.Name, "cluster", "", "name of the EKS cluster to add the nodegroup to")
+		_ = cobra.MarkFlagRequired(fs, "quickstart-profile")
+		_ = cobra.MarkFlagRequired(fs, "git-url")
+		_ = cobra.MarkFlagRequired(fs, "cluster")
+
+		cmdutils.AddRegionFlag(fs, cmd.ProviderConfig)
+		cmdutils.AddConfigFileFlag(fs, &cmd.ClusterConfigFile)
+	})
+
+	cmdutils.AddCommonFlagsForAWS(cmd.FlagSetGroup, cmd.ProviderConfig, false)
+}
+
+func doApplyGitops(cmd *cmdutils.Cmd, opts options) error {
+	if opts.quickstartNameArg == "" {
+		return errors.New("please supply a valid gitops Quick Start URL or name in --quickstart-profile")
+	}
+
+	if opts.git.URL == "" {
+		return errors.New("please supply a valid --git-url argument")
+	}
+
+	quickstartRepoURL, err := repoURLForQuickstart(opts.quickstartNameArg)
+	if err != nil {
+		return errors.Wrapf(err, "please supply a valid Quick Start name or url")
+	}
+
+	if err := cmdutils.NewGitopsMetadataLoader(cmd).Load(); err != nil {
+		return err
+	}
+	cfg := cmd.ClusterConfig
+	ctl, err := cmd.NewCtl()
+	if err != nil {
+		return err
+	}
+
+	if err := ctl.CheckAuth(); err != nil {
+		return err
+	}
+	if err := ctl.RefreshClusterConfig(cfg); err != nil {
+		return err
+	}
+	kubernetesClientConfigs, err := ctl.NewClient(cfg)
+	if err != nil {
+		return err
+	}
+	k8sConfig := kubernetesClientConfigs.Config
+
+	k8sRestConfig, err := clientcmd.NewDefaultClientConfig(*k8sConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return errors.Wrap(err, "cannot create Kubernetes client configuration")
+	}
+	k8sClientSet, err := kubeclient.NewForConfig(k8sRestConfig)
+	if err != nil {
+		return errors.Errorf("cannot create Kubernetes client set: %s", err)
+	}
+
+	// Create the flux installer. It will clone the user's repository in the outputPath
+	fluxOpts := flux.InstallOpts{
+		GitURL:      opts.git.URL,
+		GitBranch:   opts.git.Branch,
+		OutputPath:  opts.outputPath,
+		GitEmail:    opts.git.Email,
+		GitUser:     opts.git.User,
+		Namespace:   "flux",
+		GitFluxPath: "flux/",
+		Timeout:     defaultGitTimeout,
+	}
```

Comment:
`AdditionalFluxArgs: []string{"--sync-garbage-collection", "--manifest-generation"}` is missing

We should probably add them implicitly inside the flux package and not expose `AdditionalFluxArgs` externally

Reply:
Not needed since this is in the Template